server/util/networking/networking.go

```diff
@@ -236,9 +240,12 @@ func (n *HostNet) Unlock() {
 }
 
 // Get assigns a host network IP for the given VM index.
-func (a *HostNetAllocator) Get(vmIdx int) (*HostNet, error) {
+func (a *HostNetAllocator) Get() (*HostNet, error) {
+	a.mu.Lock()
+	defer a.mu.Unlock()
 	for attempt := 0; attempt < numAssignableNetworks; attempt++ {
-		netIdx := (vmIdx + attempt) % numAssignableNetworks
+		netIdx := a.idx % numAssignableNetworks
+		a.idx = (a.idx + 1) % numAssignableNetworks
 		if a.tryLock(netIdx) {
```

Comment:
nit: Maybe rename this given we're not locking anymore?

Reply:
I just inlined the logic to avoid confusion